web/handlers.go

```diff
@@ -210,6 +212,10 @@ func (h Handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		c.SetServerBusyError()
 	}
 
+	if c.Err == nil && h.IsLocal && *c.App.Config().ServiceSettings.EnableLocalMode {
```

Comment:
This looks a bit risky. Local handlers *should* all be on the local router that's not accessible remotely, but can we really trust that at no point we accidentally add them to the root router?

To address that, could we check `r.RemoteAddr` (or something similar) to make sure the request is actually local before setting a local session?

Reply:
Looks like `r.RemoteAddr` is set to `@` when the request comes from the socket, would that comparison be a good one?

Reply:
That should work fine since for normal remote connections it's `<host>:<port>`.

Reply:
Addressed in https://github.com/mattermost/mattermost-server/pull/14333/commits/d5d43ca1788baaf92fa5b5bc81c890cba4378256. Thanks @jupenur!

Reply:
@jupenur Is there any possibility to forge that value in the TCP packet header?

Reply:
@isacikgoz the TCP handshake should ensure remote address is always trustworthy